### PR TITLE
Moes branch - 01/28/2021 - Added Functions to Board.ex Module

### PR DESCRIPTION
**Pull request by mhindawash**
We added two functions to the Board module. Status function and the sunk? function. The status function checks on the status of the game inside the board, while the sunk function was just another function we needed to wrap up the statuses.

### CODE REVIEW

File: assets/package-lock.json [hunk 1]
```diff
@@ -1,9616 +1,6 @@
 {
```
Comment (herminiotorres) [inline assets/package-lock.json:1]: So @djslakivich this file you must keep, this file guarantee all the versions of node modules are lock, this is telling us everyone will have the same version of node module packages.

File: lib/barquinhos/game/board.ex [hunk 2]
```diff
@@ -25,7 +26,21 @@ defmodule Barquinhos.Game.Board do
...
   def game_over?(%__MODULE__{} = board) do
     Enum.all?(board.ships, fn ship -> Ship.sunk?(ship, board.shots) end)
   end
+
+  def status(board, shot) do
+    cond do
+      game_over?(board) -> %{board | status: :gameover}
+      sunk?(board) -> %{board | status: :sunk}
+      not hit?(board, shot) -> %{board | status: :miss}
+      true -> %{board | status: :hit}
```
Comment (herminiotorres) [inline lib/barquinhos/game/board.ex:42]:
```suggestion
      hit?(board, shot) -> %{board | status: :hit}
      true -> %{board | status: :miss}
```

File: lib/barquinhos/game/board.ex [hunk 3]
```diff
@@ -25,7 +26,21 @@ defmodule Barquinhos.Game.Board do
...
 
+  def sunk?(%__MODULE__{} = board) do
+    Enum.any?(board.ships, fn ship -> Ship.sunk?(ship, board.shots) end)
+  end
+
   def game_over?(%__MODULE__{} = board) do
     Enum.all?(board.ships, fn ship -> Ship.sunk?(ship, board.shots) end)
   end
+
+  def status(board, shot) do
```
Comment (herminiotorres) [inline lib/barquinhos/game/board.ex:37]:
```suggestion
  def status(%__MODULE__{} = board, shot) do
```

**Review by herminiotorres — APPROVED** [review]

File: assets/package-lock.json (hunk 1 above)
Comment (jacquelineIO) [reply assets/package-lock.json:1]: @herminiotorres  I put this as ignore in the .gitignore https://github.com/groxio-learning/barquinhos/blob/main/.gitignore#L50  should this be removed? (as in, it should not be in the .gitignore file) I wasn't sure if this was to be saved in the repo or not.

Comment (herminiotorres) [reply assets/package-lock.json:1]: @jacquelineIO yes, you must remove this file in the `.gitignore` and version this file again.

**Pull request merged**